lxd/daemon.go

```diff
@@ -357,6 +357,12 @@ func (d *Daemon) Authenticate(w http.ResponseWriter, r *http.Request) (trusted b
 			updateIdentityCache(d)
 		}
 
+		if len(result.IdentityProviderGroups) > 0 {
```

Comment:
There's several authentication related vars added to the request context in daemon.createCmd.
I wondered if we should be keeping all of that logic together, either by moving it into this function or by moving this out to createCmd. Let me know.